upload() in CGI.pm is broken at times

This problem was reported in:
   http://rt.cpan.org/Public/Bug/Display.html?id=12694

fileno($_) can be 0, since stdin (the usual taker of fileno 0) is closed
by Apache and can therefore be reused by the operating system when new
file descriptors are opened.  And of course, fileno($_) == 0 is false,
which means nothing gets returned.

diff -Naur --exclude=debian perl-5.8.8.orig/lib/CGI.pm perl-5.8.8/lib/CGI.pm
--- perl-5.8.8.orig/lib/CGI.pm	2006-01-09 03:40:30.000000000 +1100
+++ perl-5.8.8/lib/CGI.pm	2007-09-16 11:41:31.000000000 +1000
@@ -3411,7 +3411,7 @@
 'upload' =><<'END_OF_FUNC',
 sub upload {
     my($self,$param_name) = self_or_default(@_);
-    my @param = grep(ref && fileno($_), $self->param($param_name));
+    my @param = grep(ref && defined(fileno($_)), $self->param($param_name));
     return unless @param;
     return wantarray ? @param : $param[0];
 }
